### PR TITLE
[ADD] [15.0] purchase_warn_option

### PR DESCRIPTION
**Pull request by Shide**
This module extends the functionality of blocking messages to support more consistency
between messages and to allow you to set a blocking message quickly.

This is a PR of a series:

[base_warn_option](https://github.com/OCA/server-ux/pull/853)
[sale_warn_option](https://github.com/OCA/sale-workflow/pull/3076)
[account_warn_option](https://github.com/OCA/account-invoicing/pull/1704)
[stock_warn_option](https://github.com/OCA/stock-logistics-workflow/pull/1571)

MT-5193 @moduon @rafaelbn @yajo @EmilioPascual please review if you want :)

### CODE REVIEW

File: purchase_warn_option/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+{
+    "name": "Purchase Warn Option",
+    "summary": "Add Options to Purchase Warn Messages",
+    "version": "15.0.1.0.0",
+    "development_status": "Alpha",
+    "category": "Inventory/Purchase",
+    "website": "https://github.com/OCA/purchase-workflow",
+    "author": "Moduon, Odoo Community Association (OCA)",
+    "maintainers": ["Shide"],
```
Comment (yajo) [inline purchase_warn_option/__manifest__.py]: issue:

```suggestion
    "maintainers": ["Shide", "rafaelbn"],
```

File: purchase_warn_option/readme/CONTEXT.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+This module has been developed to maintain consistency in the messages used when
+blocking suppliers or products in purchase models.
+
+If you need this module for those reasons, these might interest you too:
+
+- base_warn_option
+- sale_warn_option
+- purchase_warn_option
```
Comment (yajo) [inline purchase_warn_option/readme/CONTEXT.md]: suggestion:

```suggestion
- sale_warn_option
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Review by yajo — APPROVED** [review]
Please take a look at the tests.

**Review by EmilioPascual — APPROVED** [review]
Functional and Code review

**Review by rafaelbn — APPROVED** [review]
/ocabot merge patch

**Comment by yajo** [comment]
@Shide please check test failures.

**Comment by Shide** [comment]
Could you please check tests?

@marcelsavegnago: purchase_default_terms_conditions
@sebastienbeau: purchase_quick
@gurneyalex: purchase_order_analytic_search

**Pull request merged**